ProperlyStopRunningPipelines.groovy

```diff
@@ -0,0 +1,102 @@
+/*
+Author: Alex Taylor
+Since: July 2018
+Description: This script stop all or a series of running pipeline jobs
+reset or became out of synch.
+Parameters: None
+Scope: Cloudbees Jenkins Platform
+*/
+
+/* This script can be used to kill off all running jobs, the latest build of a job, a specific job number, or jobs
+which have been running for a certain amount of time. This is all based on a couple of specific settings which are marked
+with comments. This script will guaranteed take 1.5 minutes to run because we want to ensure that each build command has the
+time needed to run*/
+
+import groovy.time.*;
+
+
+// A couple of example map definitions:
+
+//Used to kill all of the active jobs
+//def jobMap = [:]
+
+//Used to clean up specific builds. Put "buildNum: 0" if you want to stop the latest build
+/*def jobMap = [
```

Comment:
Why use a Map when the key is just an index? Use a normal list instead and you won't have to do all the `++jobMap.size()` everywhere.